source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRCustomRenderer.mm

```diff
@@ -46,6 +46,9 @@ - (UIView *)render:(UIView<ACRIContentHoldingView> *)viewGroup
             NSData *jsonPayload = [jsonString dataUsingEncoding:NSUTF8StringEncoding];
             ACOParseContext *context = [reg getParseContext];
             NSObject<ACOIBaseCardElementParser> *parser = [reg getCustomElementParser:type];
+            if(!parser) {
+                @throw [NSException exceptionWithName:@"ParsingError" reason:@"UnknownParser" userInfo:nil];
```

Comment:
>UnknownParser [](start = 79, length = 13)

Question: is this just a sanity check you are adding now, or were you running into issues where the parser was actually null?

Reply:
sending message to nil object is no op in objective-c; so if the parser was nil, what's returned before the change will be nil, and the caller will send message to this nil object, and it will be no op, and the effect is that the custom type element is gotten dropped.

---
In reply to: [275513041](https://github.com/Microsoft/AdaptiveCards/pull/2675#discussion_r275513041) [](ancestors = 275513041)